components/category_menu_items/category_menu_items.tsx

```diff
@@ -0,0 +1,115 @@
+import {GlobalState} from 'types/store';
+import Constants, {ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    channel: Channel;
+    intl: IntlShape;
+    openUp: boolean;
+};
+
+const CategoryMenuItems = (props: Props): JSX.Element | null => {
+    const {channel, intl, openUp} = props;
+    const dispatch = useDispatch<DispatchFunc>();
+    const displayedChannels = useSelector((state: GlobalState) => getDisplayedChannels(state));
+    const multiSelectedChannelIds = useSelector((state: GlobalState) => state.views.channelSidebar.multiSelectedChannelIds);
+    const currentTeam = useSelector((state: GlobalState) => getCurrentTeam(state));
+
+    let categories: ChannelCategory[] | undefined;
+    let currentCategory: ChannelCategory | undefined;
+
+    if (currentTeam) {
+        categories = useSelector((state: GlobalState) => getCategoriesForCurrentTeam(state));
+        currentCategory = useSelector((state: GlobalState) => getCategoryInTeamWithChannel(state, currentTeam.id, channel.id));
+    }
+
+    const moveToCategory = useCallback((categoryId: string) => {
+        dispatch(addChannelsInSidebar(categoryId, channel.id));
+        trackEvent('ui', 'ui_sidebar_channel_menu_moveToExistingCategory');
+    }, []);
+
+    const moveToNewCategory = useCallback(() => {
+        dispatch(openModal({
+            modalId: ModalIdentifiers.EDIT_CATEGORY,
+            dialogType: EditCategoryModal,
+            dialogProps: {
+                channelIdsToAdd: multiSelectedChannelIds.indexOf(channel.id) === -1 ? [channel.id] : multiSelectedChannelIds,
+            },
+        }));
+        trackEvent('ui', 'ui_sidebar_channel_menu_createCategory');
+    }, []);
+
+    const selectedChannels = multiSelectedChannelIds.indexOf(channel.id) === -1 ? [channel] : displayedChannels.filter((c) => multiSelectedChannelIds.indexOf(c.id) !== -1);
+    const allChannelsAreDMs = selectedChannels.every((selectedChannel) => selectedChannel.type === Constants.DM_CHANNEL || selectedChannel.type === Constants.GM_CHANNEL);
+    const allChannelsAreNotDMs = selectedChannels.every((selectedChannel) => selectedChannel.type !== Constants.DM_CHANNEL && selectedChannel.type !== Constants.GM_CHANNEL);
+
+    const categoryMenuItems = categories?.filter((category) => {
+        if (allChannelsAreDMs) {
+            return category.type !== CategoryTypes.CHANNELS;
+        } else if (allChannelsAreNotDMs) {
+            return category.type !== CategoryTypes.DIRECT_MESSAGES;
+        }
+        return true;
+    }).map((category: ChannelCategory) => {
+        return {
+            id: `moveToCategory-${channel.id}-${category.id}`,
+            icon: category.type === CategoryTypes.FAVORITES ? (<i className='icon-star-outline'/>) : (<i className='icon-folder-outline'/>),
+            direction: 'right' as any,
+            text: category.display_name,
+            action: () => moveToCategory(category.id),
```

Comment:
Since we're now showing the channel's current category in this list, could we add some logic to prevent clicking on the current category from triggering an API request?